### PR TITLE
fix the bugs associated with blender datas

### PR DESCRIPTION
**Pull request by KevinXu02**
fix the bug when camera.distortion_params is None, which will lead to no data when using blender data.

### CODE REVIEW

**Comment by KevinXu02** [comment]
I tested it with lego and the camera convention is not correct now......

File: nerfstudio/model_components/renderers.py [hunk 1]
```diff
@@ -50,8 +50,7 @@ def background_color_override_context(mode: Float[Tensor, "3"]) -> Generator[Non
     old_background_color = BACKGROUND_COLOR_OVERRIDE
     try:
         BACKGROUND_COLOR_OVERRIDE = mode
-        yield
-    finally:
```
Comment (KevinXu02) [inline nerfstudio/model_components/renderers.py:54]: this finally will lead to the overrided colour be overrided.

File: nerfstudio/data/dataparsers/blender_dataparser.py [hunk 2]
```diff
@@ -61,13 +62,13 @@ def __init__(self, config: BlenderDataParserConfig):
         self.data: Path = config.data
         self.scale_factor: float = config.scale_factor
         self.alpha_color = config.alpha_color
-
-    def _generate_dataparser_outputs(self, split="train"):
         if self.alpha_color is not None:
-            alpha_color_tensor = get_color(self.alpha_color)
+            self.alpha_color_tensor = get_color(self.alpha_color)
+            background_color_override_context(self.alpha_color_tensor)
```
Comment (KevinXu02) [inline nerfstudio/data/dataparsers/blender_dataparser.py:67]: override bg colour

File: nerfstudio/models/gaussian_splatting.py [hunk 3]
```diff
@@ -533,14 +533,14 @@ def get_outputs(self, camera: Cameras) -> Dict[str, Union[torch.Tensor, List]]:
         if self.training:
             # currently relies on the branch vickie/camera-grads
             self.camera_optimizer.apply_to_camera(camera)
-        if self.training:
+        # get the background color
```
Comment (KevinXu02) [inline nerfstudio/models/gaussian_splatting.py:536]: modified logic to make override on top level

**Review by brentyi — APPROVED** [review]

**Comment by KevinXu02** [comment]
It seems that my ruff formatter keeps messing everything up, but the background should work now. When using blender data the background_color should be changed to white to avoid cloulds. In the viser the rendered bg color is always overwritten to grey, it will not affect training, but I wonder if there is some way to modify it without using overwrite.

**Comment by danperazzo** [comment]
Hi everyone! How can I change the background color to white? I tried doing this, but I kept getting some strange visualizations

**Pull request merged**